python/hail/table.py

```diff
@@ -2306,7 +2306,8 @@ def collect_by_key(self, name: str= 'values') -> 'Table':
 
         Notes
         -----
-        The order of the values array is not guaranteed.
+        The order of the values array is not guaranteed. Does not work on tables
+        with `key=None`.
```

Comment:
Might be worth adding a new template in `python/hail/docs/_templates/req_keyed_table.rst` that says this method requires the table key to not be None.

Reply:
Good idea